lms/static/js/groups/views/cohort_discussions_inline.js

```diff
@@ -0,0 +1,142 @@
+            }
+        },
+
+        /**
+         Returns the html list for inline discussion topics.
+
+         Args:
+            inlineDiscussions (object): inline discussions object from server
+                with attributes 'entries', 'children' & 'subcategories'.
+
+         Returns:
+            HTML list for inline discussion topics.
+        **/
+        getInlineDiscussions: function (inlineDiscussions) {
+            var categoryTemplate = _.template($('#cohort-discussions-category-tpl').html()),
+                entryTemplate = _.template($('#cohort-discussions-subcategory-tpl').html()),
+                isCategoryCohorted = false,
+                children = inlineDiscussions.children,
+                entries = inlineDiscussions.entries,
+                subcategories = inlineDiscussions.subcategories;
+
+            return _.map(children, function (name) {
+                var html = '', entry;
+                if (entries && _.has(entries, name)) {
+                    entry = entries[name];
+                    html = entryTemplate({
+                        name: name,
+                        id: entry.id,
+                        is_cohorted: entry.is_cohorted,
+                        type: 'inline'
+                    });
+                } else { // subcategory
+                    html = categoryTemplate({
+                        name: name,
+                        entries: this.getInlineDiscussions(subcategories[name]),
+                        isCategoryCohorted: isCategoryCohorted
+                    });
+                }
+                return html;
+            }, this).join('');
+        },
+
+        /**
+         Disables the discussion category checkboxes.
+         Disables the discussion sub-category checkboxes.
+         Enables the save button for inline discussion topics.
+
+         Args:
+            disable (Bool): The flag to enable/disable the elements.
+        **/
+        changeAllInlineDiscussions: function(event) {
+            event.preventDefault();
+            this.toggleInlineDiscussions(($(event.currentTarget).prop('checked')));
+        },
+
+        /**
+         Enables the discussion category checkboxes.
+         Enables the discussion sub-category checkboxes.
+         Enables the save button for inline discussion topics.
+
+         Args:
+            disable (Bool): The flag to enable/disable the elements.
+        **/
+        changeSomeInlineDiscussions: function(event) {
+            event.preventDefault();
+            this.toggleInlineDiscussions(!($(event.currentTarget).prop('checked')));
+        },
+
+        /**
+         Enable/Disable the discussion category checkboxes.
+         Enable/Disable the discussion sub-category checkboxes.
+         Enable/Disable the save button for inline discussion topics.
+
+         Args:
+            disable (Bool): The flag to enable/disable the elements.
+        **/
+        toggleInlineDiscussions: function(disable) {
```

Comment:
You can make this `setElementsEnabled: function(enable_checkboxes, enable_save_button)` and then use it in `render()` too.